### PR TITLE
refactor: move web3

### PR DESCRIPTION
**Pull request by lelenei**
## Description

Closes # (NO_ISSUE)

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/runtime/ui/src/web3/hook/useAddEthereumChain.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    options?: RequestOptions,
+) {
+    const web3Provider = useWeb3Provider(overrides, options)
+    return useAsyncFn(async () => {
+        web3Provider.request({
+            method: EthereumMethodType.WALLET_ADD_ETHEREUM_CHAIN,
+            params: [chainDetailed, address].filter(Boolean),
+        })
+        const web3 = new Web3(web3Provider)
+        return web3.eth.addEthereumChain(chainDetailed, address, overrides)
```
Comment (guanbinrui) [inline packages/runtime/ui/src/web3/hook/useAddEthereumChain.ts]:
```suggestion
        return web3Provider.request({
            method: EthereumMethodType.WALLET_ADD_ETHEREUM_CHAIN,
            params: [chainDetailed, address].filter(Boolean),
        }, overrides)
```

File: packages/runtime/ui/src/web3/hook/useSwitchEthereumChain.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+            method: EthereumMethodType.WALLET_SWITCH_ETHEREUM_CHAIN,
+            params: [
+                {
+                    chainId: toHex(chainId),
+                },
+            ],
+        })
+
+        const web3 = new Web3(web3Provider)
+        return web3.eth.switchEthereumChain(chainId, overrides)
```
Comment (guanbinrui) [inline packages/runtime/ui/src/web3/hook/useSwitchEthereumChain.ts]:
```suggestion
        return web3Provider.request({
            method: EthereumMethodType.WALLET_SWITCH_ETHEREUM_CHAIN,
            params: [
                {
                    chainId: toHex(chainId),
                },
            ],
        }, overrides)
```

File: packages/runtime/ui/src/web3/UI/EthereumChainBoundary.tsx [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+    useRemoteControlledDialog,
+    ActionButtonPromise,
+    ActionButtonPromiseProps,
+} from '@masknet/shared'
+import { delay } from '@dimensiondev/kit'
+
+import { currentProviderSettings } from '../../plugins/Wallet/settings'
+import { WalletMessages, WalletRPC } from '../../plugins/Wallet/messages'
+import Services from '../../extension/service'
+import { useI18N } from '../../locales/i18n_generated'
```
Comment (UncleBill) [inline packages/runtime/ui/src/web3/UI/EthereumChainBoundary.tsx]:
```suggestion
import { useI18N } from '../../locales'
```

**Review by UncleBill — COMMENTED** [review]
You might create a `packages/runtime/web3` package to hold `web3-runtime`, all sub-runtime will be grouped into the final runtime.

**Comment by Jack-Works** [comment]
Toooo many merge conflicts. I guess if you still need it, starting a new PR will be much easier.

**Pull request closed without merging**